src/yunohost/backup.py

```diff
@@ -1370,22 +1355,27 @@ def copytree(src, dst, symlinks=False, ignore=None):
             restore_script = os.path.join(tmp_folder_for_app_restore, 'restore')
 
             # Restore permissions
-            if os.path.isfile(app_settings_in_archive + '/permission.ldif'):
-                filtred_entries =  ['entryUUID', 'creatorsName', 'createTimestamp', 'entryCSN', 'structuralObjectClass',
-                                    'modifiersName', 'modifyTimestamp', 'inheritPermission', 'memberUid']
-                entries = read_ldif('%s/permission.ldif' % app_settings_in_archive, filtred_entries)
-                group_list = user_group_list(['cn'])['groups']
-                for dn, entry in entries:
-                    # Remove the group which has been removed
-                    for group in entry['groupPermission']:
-                        group_name = group.split(',')[0].split('=')[1]
-                        if group_name not in group_list:
-                            entry['groupPermission'].remove(group)
-                    if not ldap.add('cn=%s,ou=permission' % entry['cn'][0], entry):
-                        raise YunohostError('apps_permission_restoration_failed',
-                                            permission=entry['cn'][0].split('.')[0],
-                                            app=entry['cn'][0].split('.')[1])
+            if os.path.isfile('%s/permissions.yml' % app_settings_new_path):
+
+                permissions = read_yaml('%s/permissions.yml' % app_settings_new_path)
+                existing_groups = user_group_list()['groups']
+
+                for permission_name, permission_infos in permissions.items():
+
+                    permission_create(permission_name, urls=permission_infos.get("urls", []))
+
+                    if "allowed" not in permission_infos:
+                        logger.warning("'allowed' key corresponding to allowed groups for permission %s not found when restoring app %s … You might have to reconfigure permissions yourself." % (permission_name, app_instance_name))
+                    else:
+                        should_be_allowed = [g for g in permission_infos["allowed"] if g in existing_groups]
+                        current_allowed = user_permission_list()["permissions"][permission_name]["allowed"]
+                        if should_be_allowed != current_allowed:
+                            user_permission_update(permission_name, remove=current_allowed, add=should_be_allowed)
+
+                os.remove('%s/permissions.yml' % app_settings_new_path)
```

Comment:
Why do we need to remove this file here ?

Reply:
No particular reason except that it's only meant for backup purpose and shouldn't later be confused by tech-savvy users as an actual permission config file

Reply:
Ok, because I was just thinking about a case when this file might be needed a other time for a restoration (by example if the restoration failed). It still just a question because I don't know exactly how work the backup here...

Reply:
A failed restoration is pretty much like a failed install, so all app files will be removed (including the setting folder). In such case, you can just re-try the restoration and this file will still be in the archive

Reply:
Ok